x-pack/plugin/search-business-rules/src/yamlRestTest/resources/rest-api-spec/test/searchbusinessrules/10_pinned_retriever.yml

```diff
@@ -0,0 +1,389 @@
+                    }
+                  }
+                },
+                {
+                  pinned: {
+                    ids: ["doc1", "doc2"],
+                    retriever: {
+                      standard: {
+                        query: {
+                          match_all: {}
+                        }
+                      }
+                    }
+                  }
+                }
+              ]
+
+  - match: { hits.total.value: 5 }
+
+---
+"pinned retriever with explicit sort on score":
+
+  - do:
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+                  sort: [ "_score" ]
+
+  - match: { hits.total.value: 5 }
+  - match: { hits.hits.0._id: doc1 }
+  - match: { hits.hits.1._id: doc2 }
+
+---
+"pinned retriever with rank window size":
+  - skip:
+      features: headers
+
+  - do:
+      headers:
+        Content-Type: application/json
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+              rank_window_size: 1
+
+  - match: { hits.total.value: 5 }
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._id: doc1 }
+  - is_true: { hits.hits.0._score > 1.0 }
+
+  - do:
+      headers:
+        Content-Type: application/json
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+              rank_window_size: 10
+
+  - match: { hits.total.value: 5 }
+  - length: { hits.hits: 5 }
+  - match: { hits.hits.0._id: doc1 }
+  - match: { hits.hits.1._id: doc2 }
+  - is_true: { hits.hits.0._score > 1.0 }
+  - is_true: { hits.hits.1._score > 1.0 }
+  - is_true: { hits.hits.2._score <= 1.0 }
+  - is_true: { hits.hits.3._score <= 1.0 }
+  - is_true: { hits.hits.4._score <= 1.0 }
+
+---
+"pinned retriever explanation":
+
+  - do:
+      headers:
+        Content-Type: application/json
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+          explain: true
+
+  - match: { hits.hits.0._id: doc1 }
+  - is_true: { hits.hits.0._explanation.value > 1.0 }
+  - is_true: { hits.hits.0._explanation.description: "pinned documents" }
+  - is_true: { hits.hits.0._explanation.description: "rank [1]" }
+  - is_true: { hits.hits.0._explanation.details.0.details.0.details.0.details.0.details.0.description: "ConstantScore" }
```

Comment:
This is where I think we can improve based on my comment above about the `RankDoc`s you are generating